geoportailv3/static/js/infobar/projectionselectordirective.js

```diff
@@ -72,11 +81,42 @@ app.ProjectionselectorDirectiveController =
     {'label': $sce.trustAsHtml('WGS84 UTM 32|31'), 'value': 'EPSG:3263*'}
       ];
       this['projection'] = this['projectionOptions'][0];
-  /** @type {ol.control.MousePostion} */
-      this['mousePositionControl'] = new ol.control.MousePosition({
-        className: 'custom-mouse-coordinates',
-        coordinateFormat: /** @type {ol.CoordinateFormatType} */
-        (goog.bind(this.mouseCoordinateFormat_, this))
+
+      let updateEl = (coordinates) => $($element)
+        .find('.custom-mouse-coordinates')
+        .text(this.mouseCoordinateFormat_(coordinates));
+
+      this['map'].on('pointermove', ngeoDebounce((e) => {
+        if (this.old3d && this.ol3d.is3dEnabled()) {
+          return;
+        }
+        updateEl(e.coordinate);
+      }, 10, false), this);
+
+      // 3D
+      let unwatch = $scope.$watch(() => ngeoOlcsService.getManager().getOl3d(), (ol3d) => {
+        if (!ol3d) {
+          return;
+        }
+        const manager = ngeoOlcsService.getManager()
+        this.ol3d = manager.getOl3d();
+
+        const scene = manager.getCesiumScene();
+        const camera = scene.camera;
+        const handler = new Cesium.ScreenSpaceEventHandler(scene.canvas);
+        const WMP = new Cesium.WebMercatorProjection(null);
+        handler.setInputAction(ngeoDebounce((movement) => {
+          if (!this.ol3d.getEnabled()) { return; }
+          let cartesian = camera.pickEllipsoid(movement.endPosition, scene.globe.ellipsoid);
+          if (!cartesian) {
+            return;
+          }
+          let cartographic = Cesium.Cartographic.fromCartesian(cartesian);
+          cartesian = WMP.project(cartographic);
+          let coordinate = [ cartesian.x, cartesian.y ];
+          updateEl(coordinate);
+        }, 10, false), Cesium.ScreenSpaceEventType.MOUSE_MOVE);
```

Comment:
This block is a copy paste from above.
I see two approaches:
- move the call to `getCoordinateFromPixel` in the control in an overridable method https://github.com/openlayers/openlayers/blob/v4.5.0/src/ol/control/mouseposition.js#L222; then extend the control and override the method with one aware of 3d;
- extend the ol.Map and override the existing `getCoordinateFromPixel` to make it aware of 3d.

The second solution is the most simple (and ambitious as it would fix the issue for all situations). I suggest we try this last one.